### PR TITLE
Master fixed

### PR DESCRIPTION
**Pull request by Paupiera**
After last rounds of tests of the workflow, I implemented some changes. If everything looks good from your side, we can merge it with master and the new release.

### CODE REVIEW

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 1]
```diff
@@ -164,7 +172,12 @@ def mv_single_ripped_nc_bins(
                     "Bin %s was ripped because of meaningless edges or pairing and afterwards no intersection was shared with any other bin so it is moved from %s to %s"
                     % (cluster, src_bin, trg_bin)
                 )
-                shutil.move(src_bin, trg_bin)
+                
+                #shutil.move(src_bin, trg_bin)
+                #shutil.copy(src_bin, trg_bin)
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:177]:
```suggestion
```

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 2]
```diff
@@ -237,24 +252,28 @@ def choose_best_ripped_bin_and_mv_if_nc(
             )
 
             # bin_A keeps the contigs
-            if cluster_A_complet > 90 and cluster_A_cont < 5:
+            if cluster_A_complet > min_comp and cluster_A_cont < max_cont:
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:255]: Another issue - not sure how important this is. If someone sets `min_comp = 0.9`, then I would expect 0.9 to also pass, so the check here should be `>=` not `>`. I.e. consider what happens if someone sets `min_comp = 1`.

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 3]
```diff
@@ -237,24 +252,28 @@ def choose_best_ripped_bin_and_mv_if_nc(
...
                 bin_A_name = cluster_A_r + ".fna"
                 src_bin = bin_path[bin_A_name]
                 trg_bin = os.path.join(
                     path_run, drep_folder, cluster_sample[cluster_A_r], bin_A_name
                 )
-                shutil.move(src_bin, trg_bin)
+                #shutil.move(src_bin, trg_bin)
+                #shutil.copy(src_bin, trg_bin)
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:262]:
```suggestion
```

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 4]
```diff
@@ -237,24 +252,28 @@ def choose_best_ripped_bin_and_mv_if_nc(
...
+            if cluster_B_r_complet > min_comp and cluster_B_r_cont < max_cont:
                 bin_B_ripped_name = cluster_B_r + "--" + cluster_A_r + ".fna"
                 bin_B_name = cluster_B_r + ".fna"
                 src_bin = os.path.join(path_bins_ripped, bin_B_ripped_name)
                 trg_bin = os.path.join(
                     path_run, drep_folder, cluster_sample[cluster_B_r], bin_B_name
                 )
-                shutil.move(src_bin, trg_bin)
+                #shutil.move(src_bin, trg_bin)
+                #shutil.copy(src_bin, trg_bin)
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:275]:
```suggestion
```

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 5]
```diff
@@ -268,24 +287,28 @@ def choose_best_ripped_bin_and_mv_if_nc(
...
+            if cluster_A_r_complet > min_comp and cluster_A_r_cont < max_cont:
                 bin_A_ripped_name = cluster_A_r + "--" + cluster_B_r + ".fna"
                 bin_A_name = cluster_A_r + ".fna"
                 src_bin = os.path.join(path_bins_ripped, bin_A_ripped_name)
                 trg_bin = os.path.join(
                     path_run, drep_folder, cluster_sample[cluster_A_r], bin_A_name
                 )
-                shutil.move(src_bin, trg_bin)
+                #shutil.move(src_bin, trg_bin)
+                #shutil.copy(src_bin, trg_bin)
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:310]:
```suggestion
```

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py [hunk 6]
```diff
@@ -384,6 +407,6 @@ def write_nc_bin_scores(
         bin_path,
         path_bins_ripped,
         bin_separator,
-        opt.comp,
-        opt.cont,
+        opt.comp*100,
+        opt.cont*100,
```
Comment (jakobnissen) [inline workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:411]: Ah, I see here is where you convert from fraction to percentage!

**Review by jakobnissen — CHANGES_REQUESTED** [review]
Looks good to me! With a few minor changes.
Would you mind if I squash your commits? Some of the commits look to be nonfunctional on their own, so they are not of much use on their own (i.e. it would cause confusing problems if we accidentally check out one of these commits). Happy to not squash if you prefer individual commits.

File: workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py (hunk 2 above)
Comment (Paupiera) [reply workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:255]: I completely agree, how can we proceed to implement the replacement of `>` by `>=` ? Should I make a new commit ?

Comment (jakobnissen) [reply workflow_avamb/src/transfer_contigs_and_aggregate_all_nc_bins.py:249]: Yes. Just make new commits - I think the best is to squash them in the end (i.e. replace all commits in this entire PR with a single commit with all changes). GitHub can do this automatically.

**Pull request merged**